pkg/terraform/provider_installer.go

```diff
@@ -38,6 +38,27 @@ func (p *ProviderInstaller) Install() (string, error) {
 	providerDir := p.getProviderDirectory()
 	providerPath := p.getBinaryPath()
 
+	_, err := os.Stat(providerDir)
+	if os.IsNotExist(err) {
+		logrus.WithFields(logrus.Fields{
+			"path": providerDir,
+		}).Debug("Provider directory destination not found, creating ...")
+		if err := os.MkdirAll(providerDir, 0755); err != nil {
+			return "", errors.Wrap(err, "can't create configuration directory")
+		}
+	}
+
+	isDirectoryWritablePath := path.Join(providerDir, ".is_directory_writable")
+	isDirectoryWritable, err := os.OpenFile(isDirectoryWritablePath, os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"path": providerDir,
+		}).Debug("Provider directory destination is not writable")
+		return "", errors.Wrap(err, "can't write in configuration directory")
+	}
+	defer isDirectoryWritable.Close()
+	defer os.Remove(isDirectoryWritablePath)
```

Comment:
Is it really needed to perform that check if the providerDir creation above succeed ? I mean if we can create the directory with necessary permissions, it's obviously writable, isn't it ?